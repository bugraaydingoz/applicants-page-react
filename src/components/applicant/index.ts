export { Applicant } from './Applicant'
export { ApplicantList } from './ApplicantList'
