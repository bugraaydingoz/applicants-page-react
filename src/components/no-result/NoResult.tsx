import React from 'react'
import { useTranslation } from 'react-i18next'

// Styles
import { H4, Body1 } from '../shared/typography'
import { Button } from '../shared/elements'

interface Props {
  onClick: () => void
}

export function NoResult({ onClick }: Props) {
  const { t } = useTranslation()
  return (
    <>
      <H4 style={{ marginBottom: 'var(--space-md)' }}>{t('noResult.title')}</H4>
      <Body1 style={{ marginBottom: 'var(--space-sm)' }}>
        {t('noResult.body')}
      </Body1>
      <Button onClick={onClick}>{t('noResult.action')}</Button>
    </>
  )
}
