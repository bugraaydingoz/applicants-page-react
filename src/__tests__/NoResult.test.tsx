import React from 'react'
import renderer from 'react-test-renderer'
import { NoResult } from '../components/no-result'

test('renders correctly', () => {
  const tree = renderer.create(<NoResult onClick={jest.fn} />).toJSON()
  expect(tree).toMatchSnapshot()
})
