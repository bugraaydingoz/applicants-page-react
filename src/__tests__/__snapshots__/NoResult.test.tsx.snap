// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h4
    className="sc-AxirZ fOnMZm"
    style={
      Object {
        "marginBottom": "var(--space-md)",
      }
    }
  >
    noResult.title
  </h4>,
  <div
    className="sc-AxhUy eCIPUw"
    style={
      Object {
        "marginBottom": "var(--space-sm)",
      }
    }
  >
    noResult.body
  </div>,
  <button
    className="sc-AxgMl bxcmaw"
    onClick={[Function]}
  >
    noResult.action
  </button>,
]
`;
